### PR TITLE
docs(control-design): add control design document

### PR DESCRIPTION
**Pull request by TakaHoribe**
## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: docs/design/autoware-architecture/control/index.md [hunk 1]
```diff
@@ -1,5 +1,108 @@
...
+
+The Autoware control system is designed as a platform for automated driving systems that can be compatible with a diverse range of vehicles.
+
+The control process in Autoware uses general information (such as target acceleration and deceleration) and no vehicle-specific information (such as brake pressure) is used. Hence it can be adjusted independently of the vehicle's drive interface enabling easy integration or performance tuning.
+
+Furthermore, significant differences that affect vehicle motion constraints, such as two-wheel steering or four-wheel steering, are addressed by switching the control vehicle model, achieving control specialized for each characteristic.
+
+Autoware's control module outputs the necessary information to control the vehicle as a substitute for a human driver. An example of the implementation, the control commands from the control module would be like the following:
```
Comment (maxime-clem) [inline docs/design/autoware-architecture/control/index.md:29]:
```suggestion
Autoware's control module outputs the necessary information to control the vehicle as a substitute for a human driver. For example, the control command from the control module looks like the following:
```

File: docs/design/autoware-architecture/control/index.md [hunk 2]
```diff
@@ -1,5 +1,108 @@
...
+- Target steering torque
+- Target speed
+- Target acceleration
+```
+
+Note that vehicle-specific values such as pedal positions and low-level information such as individual wheel rotation speeds are excluded from the command.
+
+## Vehicle Adaptation Design
+
+### vehicle interface adapter
```
Comment (maxime-clem) [inline docs/design/autoware-architecture/control/index.md:42]:
```suggestion
### Vehicle interface adapter
```

File: docs/design/autoware-architecture/control/index.md [hunk 3]
```diff
@@ -1,5 +1,108 @@
...
+
+Autoware is designed as an autonomous driving platform to be able to accommodate vehicles with various drivetrains types.
```
Comment (maxime-clem) [inline docs/design/autoware-architecture/control/index.md:44]:
```suggestion
Autoware is designed to be an autonomous driving platform able to accommodate vehicles with various drivetrain types.
```

File: docs/design/autoware-architecture/control/index.md [hunk 4]
```diff
@@ -1,5 +1,108 @@
...
+As functions related to vehicle motion, the following functions are typically installed on the vehicle side:
+
+- ABS（Anti-lock Brake System）
+- ESC（Electric Stability Control）
+
+Generally, it is difficult for an autonomous driving system to replace these ABS or ESC functions in terms of control cycles or obtainable information, and it is assumed that these functions are provided on the vehicle side. However, as an AD system, it models and controls the entire vehicle system, including these functions, regardless of whether they exist or not. Therefore, in the case of a vehicle without ABS installed, the autonomous driving system will generate a brake command while understanding that the deceleration rate will decrease during sudden braking.
+
+It should be noted that regarding these functions, there is a trade-off with the corresponding ODD. In an ODD such as slow-speed driving in a well-maintained indoor environment, it is not necessary for these functions to be installed on the vehicle.
+
+Also, note that this statement does not negate the development of ABS functionality in autonomous driving systems.
```
[the remark this answers is not in the document]
Comment (mitsudome-r) [reply docs/design/autoware-architecture/control/index.md:102]: Perhaps, something like this?
```
As functions related to vehicle motion, the following functions could be pre-installed to the vehicle:
- ABS（Anti-lock Brake System）
- ESC（Electric Stability Control）

Usually, it is difficult to turn-off or replace such features by an external autonomous driving software due to limited access to low level controllers through a drive-by-wire system.
Therefore, the control module of the AD system should be aware of the existence of underlying control functions when it calculates control command.
For example, if AD system is integrated with a vehicle without ABS, then the AD system should generate a brake command with consideration that sudden braking might cause slipping of the tires.
On the otherhand, if AD system is integrated with a vehicle with ABS, then AD system should be generating control command which realize desired vehicle behavior even with ABS modifying the command.
```

[the remark this answers is not in the document]
Comment (maxime-clem) [reply docs/design/autoware-architecture/control/index.md:102]: How about the following ?
```
ABS (Anti-lock Brake System) and ESC (Electric Stability Control) are two functions that may be pre-installed on a vehicle, directly impacting its controllability.
The control modules of Autoware assume that both ABS and ESC are installed on the vehicle and their absence may cause unreliable controls depending on the target ODD. For example, with low-velocity driving in a controlled environment, these functions are not necessary.
```

[the remark this answers is not in the document]
Comment (TakaHoribe) [reply docs/design/autoware-architecture/control/index.md]: @maxime-clem @mitsudome-r Thank you. I use @maxime-clem 's suggestion.

**Review by maxime-clem — APPROVED** [review]

**Pull request merged**